tests/end_to_end/tests.py

```diff
@@ -249,6 +250,21 @@ def step_autoconf(self):
                 auto_controller.app = app
                 auto_controller._default()
 
+    def step_add_service_endpoints(self, client_id):
+        endpoint_controller = EndpointController()
+        for security_server in self.config["security_server"]:
+            configuration = endpoint_controller.create_api_config(security_server, self.config)
+            for client in security_server["clients"]:
+                for service_description in client["service_descriptions"]:
+                    for service in service_description["services"]:
```

Comment:
Instead of this line you should use "for endpoint in service_description["endpoints"]:" as in the configuration file you have placed "endpoints" section inside "service_descriptions" section not in inside "services". The "services" loop you do not need here at all.